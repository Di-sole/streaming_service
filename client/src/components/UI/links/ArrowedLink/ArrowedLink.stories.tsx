import { Meta, StoryObj } from '@storybook/react';
import Link from './ArrowedLink';
import './ArrowedLink.module.scss';

const meta: Meta<typeof Link> = {
    title: 'Links/ArrowedLink',
    component: Link,
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof Link>;

export const ArrowedLink: Story = {
    args: {
        text: 'Ссылка',
        href: '',
    },
};
