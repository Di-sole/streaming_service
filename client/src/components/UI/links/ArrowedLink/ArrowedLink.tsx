import Image from 'next/image';
import Link from 'next/link';
import arrow from '../../../../../public/icons/arrows/arrow_right.svg';
import styles from './ArrowedLink.module.scss';

interface ArrowedLinkProps {
    text: string;
    href: string;
}

const ArrowedLink = ({ text, href }: ArrowedLinkProps) => {
    return (
        <Link className={styles.link} href={href}>
            <h3>{text}</h3>
            <Image className={styles.arrow} src={arrow} alt="" />
        </Link>
    );
};

export default ArrowedLink;
