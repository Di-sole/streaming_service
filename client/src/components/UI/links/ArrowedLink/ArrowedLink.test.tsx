import { render, screen } from '@testing-library/react';
import ArrowedLink from './ArrowedLink';

describe('ARROWEDLINK SNAPSHOTS TESTS', () => {
    test('ArrowedLink should not be changed', () => {
        render(<ArrowedLink href="/" text="text" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
