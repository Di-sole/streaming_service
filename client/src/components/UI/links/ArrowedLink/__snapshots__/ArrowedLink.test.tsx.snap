// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ARROWEDLINK SNAPSHOTS TESTS ArrowedLink should not be changed 1`] = `
<a
  class="link"
  href="/"
>
  <h3>
    text
  </h3>
  <img
    alt=""
    class="arrow"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</a>
`;
